diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -67,24 +40,50 @@ impl ToSql<Time, Sqlite> for NaiveTime {
         &self,
         out: &mut ToSqlOutput<W, Sqlite>,
     ) -> Result<IsNull, Box<Error + Send + Sync>> {
-        let s = format!("{}", self.format(SQLITE_TIME_FMT));
+        let s = self.format(SQLITE_TIME_FORMAT).to_string();
         ToSql::<Text, Sqlite>::to_sql(&s, out)
     }
 }
 
 impl FromSql<Timestamp, Sqlite> for NaiveDateTime {
     fn from_sql(value: Option<&SqliteValue>) -> Result<Self, Box<Error + Send + Sync>> {
         let text = not_none!(value).read_text();
-        let dt: Vec<&str> = text.split(' ').collect();
 
-        if dt.len() != 2 {
-            return Err("Can't parse Datetime. Too many parts. It should be in the form: YYYY-MM-DD HH:MM:SS".into());
+        let sqlite_datetime_formats = &[
+            // Most likely format
+            "%F %T%.f",
+            // Other formats in order of appearance in docs
+            "%F %R",
+            "%F %RZ",
+            "%F %R%:z",
+            "%F %T%.fZ",
+            "%F %T%.f%:z",
+            "%FT%R",
+            "%FT%RZ",
+            "%FT%R%:z",
+            "%FT%T%.f",
+            "%FT%T%.fZ",
+            "%FT%T%.f%:z",
+        ];
+
+        for format in sqlite_datetime_formats {
+            if let Ok(dt) = Self::parse_from_str(text, format) {
+                return Ok(dt);
+            }
         }
 
-        let time = NaiveTime::parse_from_str(dt[1], SQLITE_TIME_FMT)?;
-        let datetime = parse_sqlite_date(dt[0])?.and_time(time);
+        if let Ok(julian_days) = text.parse::<f64>() {
```

Comment:
Technically I believe these are all Gregorian not Julian. Chrono certainly doesn't actually deal with the Julian calendar, and sqlite [documents itself](https://sqlite.org/datatype3.html#date_and_time_datatype) as being gregorian.

I really doubt many people would be confused by the distinction unless they have to deal with Julian calendars, which is probably just historians?

Reply:
Chrono doesn't deal with julian days, but a float representing julian days is one of the documented valid date formats at https://sqlite.org/lang_datefunc.html. It's quite trivial to convert julian to gregorian, which is what this is doing.

Reply:
Oh no. You're obviously right, I should have read the whole function before I got confused by the variables.